CONTRIBUTING.md

```diff
@@ -1,39 +1,167 @@
-6. Push to the branch (`git push origin my-new-feature`)
-7. Create a new Pull Request
-8. Update [CHANGELOG.md](https://github.com/rails-api/active_model_serializers/blob/master/CHANGELOG.md)
-with a brief description of any breaking changes, fixes, features, or
-miscellaneous changes under the proper version section.
-9. Iterate on feedback given by the community (fix syntax, modify bits of code, add
+ActiveModelSerializers uses a subset of [StandardIssueLabels](https://github.com/wagenet/StandardIssueLabels) for Github Issues. You can [see our labels here](https://github.com/rails-api/active_model_serializers/labels).
+
+## Submitting a pull request (PR)
+
+1. The vast majority of development is happening under the `master` branch, currently slated for release as `0.10.x`.
+  This is where we would suggest you start.
+1. Fixing bugs is extraordinarily helpful and requires the least familiarity with ActiveModelSerializer.
+  Look for issues labeled [**Needs Bug Verification**](https://github.com/rails-api/active_model_serializers/labels/Needs%20Bug%20Verification) and [**Bug**](https://github.com/rails-api/active_model_serializers/labels/bug).
+1. Adding or fixing documentation is also fantastic!
+
+To fetch & test the library for development, do:
+
+1. Fork the repository ( https://github.com/rails-api/active_model_serializers/fork )
+1. `git clone https://github.com/{whoami}/active_model_serializers.git`
+1. `cd active_model_serializers`
+1. `bundle`
+  - To test against a particular rails version, 4.0 is usually the most buggy, set then
+      RAILS_VERSION environment variable as described in the [.travis.yml](.travis.yml).
+      e.g. `export RAILS_VERSION=4.0`.
+1. Create your PR branch (`git checkout -b my-helpful-pr`)
+1. Write tests for your feature, or regression tests highlighting a bug.
+  This is important so AcitveModel Serializers doesn't break it in a future version unintentionally.
+1. Write the feature itself, or fix your bug
+1. `bundle exec rake`
+1. Commit your changes (`git commit -am 'Add some feature'`)
+  - Use well-described, small (atomic) commits.
+1. Push to the branch (`git push origin my-helpful-pr`)
+1. Create a new Pull Request
+  - Include links to any relevant github issues.
+  - *Don't* change the VERSION file.
+  - Extra Credit: [Confirm it runs and tests pass on the rubies specified in the travis
+    config](.travis.yml). A maintainer will otherwise confirm it runs on these.
+
+1. *Bonus Points** Update [CHANGELOG.md](https://github.com/rails-api/active_model_serializers/blob/master/CHANGELOG.md)
+  with a brief description of any breaking changes, fixes, features, or
+  miscellaneous changes under the proper version section.
+1. Iterate on feedback given by the community (fix syntax, modify bits of code, add
 tests), pushing the new commits to the PR each time
 
-Remember to squash your commits and rebase off `master`.
+Remember to [squash your commits] and rebase off `master`.
```

Comment:
why is squash your commits in square brackets?

should we have brief instructions for squashing commits?